File: settings/Shedule/lesson/lesson.cpp
#include "lesson.h"

Lesson::Lesson() : enable(false), sound_begin("NONE"), sound_end("NONE")
{

}

bool Lesson::isEnable() const
{
    return enable;
}

void Lesson::setEnable(bool value)
{
    enable = value;
}

std::string Lesson::getSound_begin() const
{
    return sound_begin;
}

void Lesson::setSound_begin(const std::string &value)
{
    sound_begin = value;
}

std::string Lesson::getSound_end() const
{
    return sound_end;
}

void Lesson::setSound_end(const std::string &value)
{
    sound_end = value;
}

Time Lesson::getTime_begin() const
{
    return _time_begin;
}

void Lesson::setTime_begin(const Time &time_begin)
{
    _time_begin = time_begin;
}

Time Lesson::getTime_end() const
{
    return _time_end;
}

void Lesson::setTime_end(const Time &time_end)
{
    _time_end = time_end;
}

std::string Lesson::getTime_begin_str() const
{
    return _time_begin.toString();
}

std::string Lesson::getTime_end_str() const
{
    return _time_end.toString();
}

File: src/web_interface/js/setting_classes/lesson.js
class Lesson
{
	constructor(_enable, sound_begin, sound_end, time_begin, time_end)
	{
		this.enable = getArg(_enable, 1);
		this.sound_begin = getArg(sound_begin, 1);
		this.sound_end = getArg(sound_end, 1);
		this.time_begin = getArg(time_begin, 1);
		this.time_end = getArg(time_end, 1);
	}
	
	debug()
	{
		console.log("\t\t\t\t\tenable =", this.enable);
		console.log("\t\t\t\t\tsound_begin =", this.sound_begin);
		console.log("\t\t\t\t\tsound_end =", this.sound_end);
		console.log("\t\t\t\t\ttime_begin =", this.time_begin);
		console.log("\t\t\t\t\ttime_end =", this.time_end);
	}
}
